x-pack/plugins/fleet/server/services/agents/actions.ts

```diff
@@ -31,6 +32,8 @@ const ONE_MONTH_IN_MS = 2592000000;
 
 export const NO_EXPIRATION = 'NONE';
 
+const SIGNED_ACTIONS: Set<Partial<AgentActionType>> = new Set(['UNENROLL', 'UPGRADE']);
```

Comment:
@joeypoon By reading the linked [issue](https://github.com/elastic/ingest-dev/issues/1882) it seems that signed action should include unenroll and policy change, not upgrade.

Reply:
@aleksmaus can you chime in and confirm the actions we need signed?

Reply:
UPGRADE action is also needed to be signed (https://github.com/elastic/elastic-agent/pull/2781#issuecomment-1603399016).
In order to support upgrading the tamper protected Endpoint, the signed UPGRADE action is forwarded to Endpoint.